### PR TITLE
chore: Fix hlint warnings

### PR DESCRIPTION
**Pull request by jacereda**

### CODE REVIEW

**Comment by jacereda** [comment]
Some `hlint` warnings are dubious, but in general it's quite helpful. If there's anything annoying let me know.

**Comment by jacereda** [comment]
Looks like the Windows build is failing for some other reason...

**Comment by eriksvedang** [comment]
Yeah this is great! Let's re-trigger the build and see if it passes now...

**Review by eriksvedang — APPROVED** [review]

**Comment by eriksvedang** [comment]
@hellerve @scolsen Are you happy with this? (It touches a lot of code that you have written the bulk of, so would be great to know that you approve)

File: src/Deftype.hs [hunk 1]
```diff
@@ -480,7 +480,7 @@ genericStr pathStrings originalStructTy@(StructTy (ConcreteNameTy typeName) _) m
                in concatMap
                     (depsOfPolymorphicFunction typeEnv env [] "prn" . typesStrFunctionType typeEnv env)
                     (remove isFullyGenericType (map snd memberPairs))
-                    ++ (if isTypeGeneric concreteStructTy then [] else [defineFunctionTypeAlias ft])
+                    ++ [defineFunctionTypeAlias ft | not (isTypeGeneric concreteStructTy)]
```
[the remark this answers is not in the document]
Comment (jacereda) [reply src/Deftype.hs:483]: I guess it's a list comprehension with guard.

**Review by hellerve — APPROVED** [review]
I concede to the wisdom of the linter!

**Comment by eriksvedang** [comment]
Needs some rebasing now @jacereda 

Will merge ASAP.

**Comment by scolsen** [comment]
Hey! Some of these transforms are pretty smart. hlint is good :O

**Comment by jacereda** [comment]
Rebased.

**Pull request merged**